### PR TITLE
[17.0][MIG] analytic_partner: Migration to 17.0

### PR DESCRIPTION
**Pull request by BernatObrador**
Module migrated to version 17.0

cc https://github.com/APSL 160220

@miquelalzanillas @lbarry-apsl @mpascuall @peluko00 @javierobcn @ppyczko please review

### CODE REVIEW

File: analytic_partner/views/res_partner_views.xml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <odoo>
     <record model="ir.actions.act_window" id="act_analytic_cost_revenue">
         <field name="context">{'search_default_group_date': 1}</field>
-        <field name="domain">[('other_partner_id', 'child_of', active_id)]</field>
+        <field
+            name="domain"
+        >[('other_partner_id', 'child_of', context.get('active_id'))]</field>
```
[the remark this answers is not in the document]
Comment (BernatObrador) [reply analytic_partner/views/res_partner_views.xml:7]: Id it's not working at the domain. Odoo shows and error saying id does not exist.
The MR that's shows how to change active_id, active_ids and active_model uses this:
https://github.com/odoo/odoo/commit/daf05d48ac76d500aa1285184bdddc4c67641d58#:~:text=context%3D%22%7B%27default_access_mode%27%3A%20%27edit%27%2C%20%27default_project_id%27%3A%20context.get(%27active_id%27)%7D%22%20display%3D%22always%22/%3E

[the remark this answers is not in the document]
Comment (BernatObrador) [reply analytic_partner/views/res_partner_views.xml:7]: Nvm, with 'search_default_filter_domain': "[('other_partner_id', 'child_of', 'id')]" looks it's working well.

**Comment by BernatObrador** [comment]
@pedrobaeza You know how to change active_id from here: https://github.com/OCA/account-analytic/pull/686/files#diff-f861e008ee81bfd994897fefa1fe928a7d19889d6dc187006a19dee44dbb6dab:~:text=%3Cfield%20name%3D%22domain%22%3E%5B(%27other_partner_id%27%2C%20%27child_of%27%2C%20active_id)%5D%3C/field%3E
When changing this to id odoo shows that error: EvalError: Can not evaluate python expression: ([('other_partner_id', 'child_of', id)])
Error: Name 'id' is not defined.
So maybe for now it's now a way to change active_id at the domain?

**Comment by pedrobaeza** [comment]
But is there any warning using `active_id`? You should keep it if not.

**Comment by BernatObrador** [comment]
@pedrobaeza thanks! It's not raising any warnings so, it's ok.

**Review by ppyczko — APPROVED** [review]
LGTM, tested in runboat

**Review by mpascuall — APPROVED** [review]
LGTM, tested in runboat

**Review by peluko00 — COMMENTED** [review]
LGTM, tested in runboat

**Review by peluko00 — APPROVED** [review]
LGTM, tested in runboat

**Review by pedrobaeza — APPROVED** [review]
/ocabot migration analytic_partner
/ocabot merge nobump

**Pull request merged**